src/MSBuild/XMake.cs

```diff
@@ -1335,7 +1344,20 @@ string[] commandLine
                         messagesToLogInBuildLoggers = GetMessagesToLogInBuildLoggers(commandLineString);
                     }
 
-                    buildManager.BeginBuild(parameters, messagesToLogInBuildLoggers);
+                    // Log a message for every response file and include it in log
+                    foreach (var responseFilePath in s_includedResponseFiles)
```

Comment:
It probably doesn't matter too much just because there probably aren't too many response files, but this is probably faster if you add to lists. (I'm happy trying to verify or falsify that with you if you'd like to see how Benchmark.NET works!)

Reply:
I definitely agree here. Collect the initial list with `.ToList()`, then add to it in this loop. I think it can continue to be a local variable, too.

Reply:
Ping on this 🙂